packages/playback-core/src/index.ts

```diff
@@ -332,12 +358,74 @@ export const loadMedia = (
     | 'liveSyncPosition'
     | 'subtitleTracks'
     | 'subtitleTrack'
+    | 'userConfig'
   >
 ) => {
   const shouldUseNative = useNative(props, mediaEl);
   const { src } = props;
   if (mediaEl && shouldUseNative) {
     if (typeof src === 'string') {
+      const getStreamInfo = () => {
```

Comment:
This method seems like it should be in a separate module and accepts a URL to fetch. Even better would be to separate it out into multiple methods. Methods that do the fetch and methods that do the data processing. It would then make writing tests for the data processing methods a lot easier, since they won't be all inline and could short-circuit the fetching with hard-coded data.

Reply:
@gkatsev Thoughts on sharing code between this and the hls.js permutation? There's def overlap, but folks may find the functions a bit too abstracted at that point?